R/assertions.R

```diff
@@ -12,18 +12,14 @@ on_failure(is_input_id) <- function(call, env) {
 
 
 is_header <- function(x) {
-  is.null(x) || (is.character(x) && length(x) == 1 && !is.na(x))
+  is.null(x) || length(x) == 1 && (is.na(x) || is.character(x))
```

Comment:
Self reassurance that the order is kept as we expect it to be

```suggestion
  is.null(x) || (length(x) == 1 && (is.na(x) || is.character(x)))
```